src/manifest.c

```diff
@@ -616,16 +616,27 @@ struct list *create_update_list(struct manifest *current, struct manifest *serve
 		file = list->data;
 		list = list->next;
 
-		/* Skip files that aren't updated */
-		fullname = mk_full_filename(path_prefix, file->filename);
-		if (fullname == NULL) {
-			abort();
-		}
-		if (verify_file(file, fullname)) {
+		if (file->peer) {
+			/* Skip files that aren't updated, but only if they are
+			 * installed in the current version (tracked). */
+			if (file->is_tracked &&
+			    hash_equal(file->hash, file->peer->hash)) {
```

Comment:
I don't think we want to continue here yet because we also need to verify_file on the installed file first.

The file->is_tracked and hash_equal should be prerequisites for entering the verify_file block however.